tsdb/index/postings.go

```diff
@@ -409,6 +410,10 @@ type Postings interface {
 	Err() error
 }
 
+type AdditionalPostingsProvider interface {
```

Comment:
I would suggest renaming this to `PartitionPostingsFunc`, to be in line with `VerticalChunkSeriesMergeFunc`.